### PR TITLE
Language changer

### PR DESCRIPTION
**Pull request by Cs4K1Sr4C**

### CODE REVIEW

File: src/utils/prompts.ts [hunk 1]
```diff
@@ -15,20 +15,52 @@ export const createModel = (settings: ModelSettings) =>
...
+export const createTasksPrompt = new PromptTemplate({
+  template:
+    "You are You are an AI task creation agent and your objective is: `{goal}`. You have the following incomplete tasks `{tasks}` and have just executed the following task `{lastTask}` and received the following solution `{result}`. Based on this, create a new task to be completed by your AI system ONLY IF NEEDED such that your goal is more closely reached or completely reached. Use the `{customLanguage}` language to create the new task. Respond only with an Array of Strings which has the following syntax:`[the remaining and appropriate new task or tasks on the language you have to use]`",
+  inputVariables: ["goal", "tasks", "lastTask", "result", "customLanguage"],
+});*/
+
+
 export const startGoalPrompt = new PromptTemplate({
   template:
     "You are an autonomous task creation AI called AgentGPT. You have the following objective `{goal}`. Create a list of zero to three tasks to be completed by your AI system such that your goal is more closely reached or completely reached. Return the response as an array of strings that can be used in JSON.parse()",
```
Comment (awtkns) [inline src/utils/prompts.ts:39]: You will probably need to add something to prompt like: `Respond in language `{customLanguage}`

File: src/utils/prompts.ts [hunk 2]
```diff
@@ -15,20 +15,52 @@ export const createModel = (settings: ModelSettings) =>
     maxTokens: 400,
   });
 
+/*export const startGoalPrompt = new PromptTemplate({
+  template:
+    "You are an autonomous task creation AI called AgentGPT. You have the following objective `{goal}`. Create a list of zero to three tasks to be completed by your AI system such that your goal is more closely reached or completely reached. Use the `{customLanguage}` language and respond only with an Array by the following syntax:`[1...3 tasks on the language you have to use]`",
+  inputVariables: ["goal", "customLanguage"],
+});
+
+export const executeTaskPrompt = new PromptTemplate({
```
Comment (awtkns) [inline src/utils/prompts.ts:24]: the old promts can be deleted if the new one is working

File: src/utils/languages.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+export const languages = [
+  { code: "en", name: "English", flag: "🇺🇸" },
```
Comment (awtkns) [inline src/utils/languages.ts:2]: nice!

File: src/pages/_app.tsx [hunk 4]
```diff
@@ -8,15 +8,21 @@ import "../styles/globals.css";
...
   return (
     <SessionProvider session={session}>
-      <Analytics />
-      <GoogleAnalytics trackPageViews />
-      <Component {...pageProps} />
+      <I18nextProvider i18n={i18n}>
+        <Analytics />
+        <GoogleAnalytics trackPageViews />
+        <Component {...pageProps} />
+      </I18nextProvider>
```
Comment (awtkns) [inline src/pages/_app.tsx:25]:
```suggestion
       <Analytics />
       <GoogleAnalytics trackPageViews />
      <I18nextProvider i18n={i18n}>
        <Component {...pageProps} />
      </I18nextProvider>
```

File: src/middleware.ts [hunk 5]
```diff
@@ -22,10 +23,12 @@ async function shouldRateLimit(request: NextRequest): Promise<boolean> {
   return !success;
 }
 
-const rateLimitedResponse = () =>
-  new Response("Too many requests, please try again later.", {
+const rateLimitedResponse = () => {
+  const [ t ] = useTranslation();
```
Comment (awtkns) [inline src/middleware.ts:27]: Are we able to use hooks outside of the react context? Not sure we need to translate this as we only look at the error code in the front end.

File: src/i18n.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import i18n from "i18next";
```
Comment (awtkns) [inline src/i18n.js:1]: is it possible to make this a ts file?

File: src/components/AutonomousAgent.ts [hunk 7]
```diff
@@ -21,28 +22,33 @@ class AutonomousAgent {
...
   ) {
     this.name = name;
     this.goal = goal;
     this.renderMessage = renderMessage;
     this.shutdown = shutdown;
     this.modelSettings = modelSettings;
+    this.customLanguage = customLanguage;
     this.session = session;
     this._id = v4();
+    this.t = useTranslation();
```
Comment (Dogtiti) [inline src/components/AutonomousAgent.ts:51]: This may not work as hooks can only be called inside of the body of a function component.

**Comment by awtkns** [comment]
@Dogtiti noticed https://github.com/Dogtiti/AutoGPT-Next-Web. Did you maybe want to combine forces, can we bring all changes over to this main repo?

**Comment by Dogtiti** [comment]
@awtkns I'm doing that right now. On the one hand, I maintain a separate project for some specific scenarios, and on the other hand, I will contribute some features or ideas to AgentGPT that it does not have.

**Comment by Dogtiti** [comment]
@awtkns Actually, I am more oriented towards the Chinese community, and I will make some modifications that are more in line with the habits of the Chinese community. If you find these changes interesting, we can also synchronize them. That's why I am not only contributing to AgentGPT, because AgentGPT is aimed at all user groups and needs to consider everyone's habits. By the way, AgentGPT is a great project.

**Comment by awtkns** [comment]
@Dogtiti would it be possible for us to connect on discord?

**Comment by Dogtiti** [comment]
@awtkns I'd be happy to. Could you provide your Discord for me, I and my colleagues would like to communicate with you

**Comment by awtkns** [comment]
> @awtkns I'd be happy to. Could you provide your Discord for me, I and my colleagues would like to communicate with you

@Dogtiti the link is in the readme! Message me on there and I can add you to the dev group!

**Comment by Dogtiti** [comment]
@awtkns Get, I've send a message to you

**Comment by awtkns** [comment]
> @awtkns
> Do you mean that you want to merge the repository from @Dogtiti and vica versa?

@Cs4K1Sr4C nope, the changes are still good here 😀

File: next-i18next.config.js [hunk 8]
```diff
@@ -22,7 +22,7 @@ module.exports = {
     ],
     defaultLocale: "en",
   },
-  localePath: typeof window === "undefined" ? "./public/locales" : "/locales",
+  localePath: typeof window === "undefined" ? "./public/locales" : "./locales",
```
Comment (Dogtiti) [inline next-i18next.config.js:25]: `/locales` means `pulice/locales` in next.js `/locales` equals to get fiels in publice folder ,there is an example https://github.com/i18next/next-i18next/blob/master/examples/simple/next-i18next.config.js

**Pull request closed without merging**